src/app/client/src/app/modules/shared/services/navigation-helper/navigation-helper.service.ts

```diff
@@ -234,4 +234,15 @@ export class NavigationHelperService {
       this.router.navigate(['/resources']);
     }
   }
+
+  /* Used for removing DF routes and duplicate routes from history which will cause looping */
+  removeDfRoutes() {
+    this._history = this._history.filter(h => !(h.url.includes('discussion-forum')))
```

Comment:
Navigation helper service should not have any code specific to any component(like her we are using 'discussion-forum' string). We should not add any checks like this in this server.

Find the actual root cause for failure.